tracer/src/Datadog.Trace/ClrProfiler/AutoInstrumentation/GraphQL/GraphQLSecurityCommon.cs

```diff
@@ -0,0 +1,116 @@
+// <copyright file="GraphQLSecurityCommon.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+
+using System;
+using System.Collections.Generic;
+using System.Web;
+using Datadog.Trace.AppSec;
+using Datadog.Trace.AppSec.Coordinator;
+
+namespace Datadog.Trace.ClrProfiler.AutoInstrumentation.GraphQL;
+
+internal sealed class GraphQLSecurityCommon
+{
+    private static GraphQLSecurityCommon _instance;
+
+    private readonly Dictionary<IScope, Dictionary<string, object>> _scopeResolvers;
+
+    private GraphQLSecurityCommon()
+    {
+        _scopeResolvers = new();
+    }
+
+    private static GraphQLSecurityCommon GetInstance()
+    {
+        return _instance ??= new GraphQLSecurityCommon();
```

Comment:
This isn't atomic, so could lead to multiple instances being created if two threads call it at the same time. Probably `Lazy<T>` is the way to go.